tests/ui/invalid_pyclass_args.stderr

```diff
@@ -1,223 +1,356 @@
-  --> tests/ui/invalid_pyclass_args.rs:24:20
+  --> tests/ui/invalid_pyclass_args.rs:25:20
    |
-24 | #[pyclass(module = my_module)]
+25 | #[pyclass(module = my_module)]
    |                    ^^^^^^^^^
 
-error: expected one of: `crate`, `dict`, `eq`, `eq_int`, `extends`, `freelist`, `frozen`, `get_all`, `hash`, `mapping`, `module`, `name`, `ord`, `rename_all`, `sequence`, `set_all`, `subclass`, `unsendable`, `weakref`
-  --> tests/ui/invalid_pyclass_args.rs:27:11
+error: expected one of: `crate`, `dict`, `eq`, `eq_int`, `extends`, `freelist`, `frozen`, `get_all`, `hash`, `mapping`, `module`, `name`, `ord`, `rename_all`, `sequence`, `set_all`, `str`, `subclass`, `unsendable`, `weakref`
+  --> tests/ui/invalid_pyclass_args.rs:28:11
    |
-27 | #[pyclass(weakrev)]
+28 | #[pyclass(weakrev)]
    |           ^^^^^^^
 
 error: a `#[pyclass]` cannot be both a `mapping` and a `sequence`
-  --> tests/ui/invalid_pyclass_args.rs:31:8
+  --> tests/ui/invalid_pyclass_args.rs:32:8
    |
-31 | struct CannotBeMappingAndSequence {}
+32 | struct CannotBeMappingAndSequence {}
    |        ^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 error: `eq_int` can only be used on simple enums.
-  --> tests/ui/invalid_pyclass_args.rs:52:11
+  --> tests/ui/invalid_pyclass_args.rs:53:11
    |
-52 | #[pyclass(eq_int)]
+53 | #[pyclass(eq_int)]
    |           ^^^^^^
 
 error: The `hash` option requires the `frozen` option.
-  --> tests/ui/invalid_pyclass_args.rs:59:11
+  --> tests/ui/invalid_pyclass_args.rs:60:11
    |
-59 | #[pyclass(hash)]
+60 | #[pyclass(hash)]
    |           ^^^^
 
 error: The `hash` option requires the `eq` option.
-  --> tests/ui/invalid_pyclass_args.rs:59:11
+  --> tests/ui/invalid_pyclass_args.rs:60:11
    |
-59 | #[pyclass(hash)]
+60 | #[pyclass(hash)]
    |           ^^^^
 
 error: The `ord` option requires the `eq` option.
-  --> tests/ui/invalid_pyclass_args.rs:74:11
+  --> tests/ui/invalid_pyclass_args.rs:75:11
    |
-74 | #[pyclass(ord)]
+75 | #[pyclass(ord)]
    |           ^^^
 
+error: invalid format string: expected `'}'` but string was terminated
+  --> tests/ui/invalid_pyclass_args.rs:98:19
+   |
+98 | #[pyclass(str = "{")]
+   |                  -^ expected `'}'` in format string
+   |                  |
+   |                  because of this opening brace
+   |
+   = note: if you intended to print `{`, you can escape it using `{{`
+
+error: invalid format string: expected `'}'`, found `'$'`
+   --> tests/ui/invalid_pyclass_args.rs:102:19
+    |
+102 | #[pyclass(str = "{$}")]
+    |                  -^ expected `'}'` in format string
+    |                  |
+    |                  because of this opening brace
+    |
+    = note: if you intended to print `{`, you can escape it using `{{`
+
+error: The optional string format shorthand argument to `str` is incompatible with any renaming via `name` or `rename_all`.  You should remove the string format shorthand argument and implement the `Display` trait or implement `__str__` directly.
```

Comment:
I wonder whether we should just use ``The format string syntax is incompatible with any renaming via `name` or `rename_all`.`` for this error message... Because to implement `__str__` directly the `str` options needs to be completely removed, not just the shorthand argument, and there is also the option of removing any renaming. I can't really find a good way to accurately put all cases in a compact suggestion here.

Reply:
Maybe we should instead include an issue number with the relevant discussion?

Reply:
That is a good point, we can leave it simply as the error.  I will make that change.  As for linking to an issue number, we pretty much had all of the discussions on this pull request.  Should I link to this PR then?

Reply:
It's probably enough to have the link to the PR in the docs.